pkg/proxy/proxy.go

```diff
@@ -496,43 +505,67 @@ func (r *RuntimeProxy) containerStats(ctx context.Context, method string, req, r
 	return resp, nil
 }
 
-func (r *RuntimeProxy) handleImage(ctx context.Context, method string, req, resp CRIObject) (interface{}, error) {
+// We don't want to force the user to prefix image names so instead, prefer
+// to say the image is not present if it's not available to all CRIs
+func (r *RuntimeProxy) handleImageStatus(ctx context.Context, method string, req, resp CRIObject) (interface{}, error) {
 	in := req.(ImageObject)
-	client, unprefixed, err := r.clientForImage(in.Image(), true)
-	if client == nil {
-		// the client is offline
-		return resp, nil
+	for i := range r.clients {
+		client, err := r.clientAtIndex(i)
+		if err != nil {
+			continue
+		}
+		imageName := in.Image()
+		_, err = client.invokeWithErrorHandling(ctx, method, req, resp)
+		if err != nil {
+			glog.Errorf("Error in ImageStatus for client %s: %v", client.getID(), err)
+			return nil, err
+		}
+		if out, ok := resp.(ImageStatusResponse); ok && out.Image() != nil {
```

Comment:
Nitpicking, but since this handler is only for ImageStatus(), the cast should always succeed.

Reply:
For this one and most things like this where there's a whole bunch of things involved that might have an error in them, I'd rather be defensive.